pdata/pmetric/json.go

```diff
@@ -64,3 +68,603 @@ func (d *jsonUnmarshaler) UnmarshalMetrics(buf []byte) (Metrics, error) {
 	otlp.MigrateMetrics(md.ResourceMetrics)
 	return internal.MetricsFromProto(md), nil
 }
+
+type jsoniterUnmarshaler struct {
+}
+
+// NewJSONiterUnmarshaler returns a model.Unmarshaler. Unmarshals from OTLP json bytes.
+func NewJSONiterUnmarshaler() Unmarshaler {
+	return newJSONITERUnmarshaler()
+}
+
+func newJSONITERUnmarshaler() *jsoniterUnmarshaler {
+	return &jsoniterUnmarshaler{}
+}
+
+func (d *jsoniterUnmarshaler) UnmarshalMetrics(buf []byte) (Metrics, error) {
+	iter := jsoniter.ConfigFastest.BorrowIterator(buf)
+	defer jsoniter.ConfigFastest.ReturnIterator(iter)
+	td := readMetricsData(iter)
+	err := iter.Error
+	if err != nil {
+		return Metrics{}, err
+	}
+	return internal.MetricsFromProto(td), err
+}
+
+func readMetricsData(iter *jsoniter.Iterator) otlpmetrics.MetricsData {
+	td := otlpmetrics.MetricsData{}
+	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+		switch f {
+		case "resource_metrics", "resourceMetrics":
+			iter.ReadArrayCB(func(iterator *jsoniter.Iterator) bool {
+				td.ResourceMetrics = append(td.ResourceMetrics, readResourceMetrics(iter))
+				return true
+			})
+		default:
+			iter.ReportError("readMetricsData", fmt.Sprintf("unknown field:%v", f))
+		}
+		return true
+	})
+	return td
+}
+
+func readResourceMetrics(iter *jsoniter.Iterator) *otlpmetrics.ResourceMetrics {
+	rs := &otlpmetrics.ResourceMetrics{}
+	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+		switch f {
+		case "resource":
+			iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+				switch f {
+				case "attributes":
+					iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+						rs.Resource.Attributes = append(rs.Resource.Attributes, readAttribute(iter))
+						return true
+					})
+				case "droppedAttributesCount", "dropped_attributes_count":
+					rs.Resource.DroppedAttributesCount = iter.ReadUint32()
+				default:
+					iter.ReportError("readResourceMetrics.resource", fmt.Sprintf("unknown field:%v", f))
+				}
+				return true
+			})
+		case "instrumentationLibraryMetrics", "instrumentation_library_metrics", "scopeMetrics", "scope_metrics":
```

Comment:
Remove `instrumentationLibraryMetrics` and `instrumentation_library_metrics`. We dropped that support.